google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleFrameworkDetector.java

```diff
@@ -66,8 +76,37 @@ public FileType getFileType() {
   @NotNull
   @Override
   public ElementPattern<FileContent> createSuitableFilePattern() {
-    return StandardPatterns.or(
-        FileContentPattern.fileContent().withName("app.yaml"),
-        FileContentPattern.fileContent().withName("app.yml"));
+    return new AppEngineFlexPattern().withAppEngineFlexYAMLContent();
+  }
+
+  /**
+   * IntelliJ API pattern class that checks for App Engine Flex project file presence and checks it
+   * has required configuration lines to avoid spurious detection.
+   */
+  private static class AppEngineFlexPattern
```

Comment:
nit: Use `final` by default to prevent unwanted inheritance, or document for inheritance if it's expected: http://go/java-practices/final